src/v0/destinations/mp/util.js

```diff
@@ -322,6 +325,80 @@ const combineBatchRequestsWithSameJobIds = (inputBatches) => {
   return combineBatches(combineBatches(inputBatches));
 };
 
+const parseConfigArray = (arr, key) => {
+  if (!arr) {
+    return [];
+  }
+  return arr.map((item) => item[key]);
+};
+
+/**
+ * Trims the traits and contextTraits objects based on the setOnceProperties array and returns an object containing the modified traits, contextTraits, and setOnce properties.
+ *
+ * @param {object} traits - An object representing the traits.
+ * @param {object} contextTraits - An object representing the context traits.
+ * @param {string[]} setOnceProperties - An array of property paths to be considered for the setOnce transformation.
+ * @returns {object} - An object containing the modified traits, contextTraits, and setOnce properties.
+ *
+ * @example
+ * const traits = { name: 'John', age: 30 };
+ * const contextTraits = { country: 'USA', language: 'English', address: { city: 'New York', state: 'NY' }}};
+ * const setOnceProperties = ['name', 'country', 'address.city'];
+ *
+ * const result = trimTraits(traits, contextTraits, setOnceProperties);
+ * console.log(result);
+ * // Output: { traits: { age: 30 }, contextTraits: { language: 'English' }, setOnce: { $name: 'John', $country_code: 'USA', city: 'New York'} }
+ */
+function trimTraits(traits, contextTraits, setOnceProperties) {
+  let sentOnceTransform;
```

Comment:
setOnceTransform??

Reply:
changed it to `sentOnceTransformedPayload`

Reply:
I mean, sent or set?

Reply:
no "set". Because the name of the function itself is `setOnce`